### PR TITLE
feat(native_transform): add schema bridge for fitted transform DAGs (PR B7)

### PR DESCRIPTION
**Pull request by sallycr**
## Context

This is PR B7 of the `native_transform` migration split (B1-B8), and the last sub-PR before PR C (Ray compute lib). B1-B6 and B8 are already merged (#1570, #1603, #1683, #1684, #1685, #1730, #1729).

Once a `TransformSpec` DAG (PR B6) is fitted and materialized into a `TorchTransformModule`, something needs to describe that module's input/output contract as a `ModelSchema` — the shape/dtype metadata Triton config generation and served-model validation consume. This PR adds that bridge.

## Changes

- New `michelangelo/lib/native_transform/torch/schema.py`:
  - `derive_native_transform_schema(transform_spec, transform_module, sample_data=None)`: the public entry point. Filters output columns via the spec's `columns_to_keep`, optionally runs `sample_data` through the module to infer shapes/dtypes, and assembles a `ModelSchema`.
  - Supporting helpers: `_to_batched_tensor` (sample value -> batched tensor), `_get_sample_shapes_and_dtypes` (forward-pass-driven inference), `_collect_input_shape_overrides` / `_validate_input_shape_overrides` (the spec's manual `input_shape` overrides, needed for columns a forward pass can't pin — e.g. a ragged array's fixed max length, or an all-null sample column), `_build_schema_items` / `_filter_output_cols` / `_create_native_transform_schema` (schema assembly with a documented dtype/shape merge priority: spec declarations win over forward-pass inference).
  - Imports `DataType`, `ModelSchema`, `ModelSchemaItem` from the existing `michelangelo.lib.model_manager.schema` package rather than redefining them.
- New `michelangelo/lib/native_transform/torch/tests/test_schema.py`: pytest-style tests covering every function, mirroring the source repo's equivalent test coverage (shape/dtype inference across dtype combinations including int/float promotion, null-column handling, `columns_to_keep` filtering, the `input_shape` override guard and its misplaced-override error path, and full end-to-end `derive_native_transform_schema` runs).

## Design notes

- Preserves the internal implementation's exact dtype/shape merge semantics (a spec-declared `output_dtype` always overrides a forward-pass-derived dtype, even in the one internal-only case where `output_dtype` isn't a real torch dtype — kept as-is for behavioral parity, since no current layer spec in this codebase actually hits that path).
- No changes to `transform_spec.py`, `base_transform_module.py`, or any other B1-B6/B8 file — this PR only adds new files and imports from what's already merged.

## Testing

- `pytest michelangelo/lib/native_transform/` (full suite, not just the new file): 433 passed, 0 failed, no regressions.
- `ruff format --check` and `ruff check`: both clean on the new files.
- Coverage: automated `coverage run` segfaults on this torch-importing target in the sandbox test venv (a known environment limitation, not specific to this PR); coverage was instead verified by manual traceability — every function in `schema.py` has at least one dedicated test class, and every branch (derived vs. spec-declared dtype/shape, override validation, null-sample fallback, `columns_to_keep` filtering with and without exclusions) has an explicit assertion.

## Scope

This PR is self-contained and does not overlap with PR C (Ray compute lib), which starts only after this merges.

### CODE REVIEW

File: python/michelangelo/lib/native_transform/torch/schema.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+        A tuple of ``(batched_tensor, shape)``, where ``shape`` excludes the
+        batch dimension. Returns a zero-filled fallback tensor and ``None``
+        shape when ``value`` is ``None`` or fails conversion.
+    """
+    fallback = torch.zeros([1, 1], dtype=dtype_hint or torch.float32)
+    if value is None:
+        return fallback, None
+    try:
+        tensor = torch.as_tensor(value)
+    except (TypeError, ValueError):
```
Comment (kechen2013) [inline python/michelangelo/lib/native_transform/torch/schema.py]: Wondering if this should stay `except Exception` like internal? `torch.as_tensor` raises `RuntimeError` (not `TypeError`/`ValueError`) for a list with a null element, `Decimal`, `datetime`, or a dict, so a `TensorColFillNone` column sampling `[1.0, None, 3.0]` now propagates out of `derive_native_transform_schema` instead of falling back to the zero-filled tensor the docstring promises. The output forward pass just below still catches broadly.

Comment (sallycr) [reply python/michelangelo/lib/native_transform/torch/schema.py]: thanks for catching this. updated with Exception and adding the tests.

**Review by kechen2013 — APPROVED** [review]
